### PR TITLE
feat(DWS): refactor resource_huaweicloud_dws_cluster to new SDK

### LINKED ISSUE #1520: [DWS] Add DWS service API in golang SDK
Add DWS service API in golang SDK

### Proposal

those api will be used by 
- #1505

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:
 refactor resource_huaweicloud_dws_cluster to new SDK

**Which issue this PR fixes**:

fixes #1520

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dws' TESTARGS='-run=TestAccResourceDWS_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dws -v -run=TestAccResourceDWS_basic -timeout 360m -parallel 4
=== RUN   TestAccResourceDWS_basic
=== PAUSE TestAccResourceDWS_basic
=== CONT  TestAccResourceDWS_basic
--- PASS: TestAccResourceDWS_basic (1861.52s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dws       1862.038s
```

### CODE REVIEW

File: huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go [hunk 1]
```diff
@@ -0,0 +1,574 @@
...
+		ReadContext:   resourceDwsClusterRead,
+		DeleteContext: resourceDwsClusterDelete,
+		UpdateContext: resourceDwsClusterUpdate,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+			Update: schema.DefaultTimeout(60 * time.Minute),
```
Comment (niuzhenguo) [inline huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go:31]: All Timeouts are 60m, better to use an appropriate value, and need to update docs as well.

Comment (hmbig2) [reply huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go:31]: OK, docs has been updated.  All operation time exceeds 40 minutes, so keep it as it is

File: huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_cluster_test.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+func testAccDwsCluster_basic(rName string, numberOfNode int, publicIpBindType string, password string) string {
+	baseResource := testAccBaseResource(rName)
+
+	return fmt.Sprintf(`
+%s
+
+resource "huaweicloud_dws_cluster" "test" {
+  name = "%s"
+  node_type = "dws.m3.xlarge"
```
Comment (chengxiangdong) [inline huaweicloud/services/acceptance/dws/resource_huaweicloud_dws_cluster_test.go]: the TF code need to be formated

File: huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go [hunk 3]
```diff
@@ -0,0 +1,574 @@
...
+				ForceNew: true,
+			},
+
+			"user_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"user_pwd": {
```
Comment (chengxiangdong) [inline huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go:77]: pls add `Sensitive: true` for secret fields

File: huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go [hunk 4]
```diff
@@ -0,0 +1,574 @@
...
+		return fmtp.DiagErrorf("error creating DWS v1 client, err=%s", err)
+	}
+
+	opts := &cluster.CreateOpts{
+		Name:                d.Get("name").(string),
+		AvailabilityZone:    d.Get("availability_zone").(string),
+		SubnetID:            d.Get("network_id").(string),
+		NodeType:            d.Get("node_type").(string),
+		UserName:            d.Get("user_name").(string),
+		UserPwd:             d.Get("user_pwd").(string),
```
Comment (chengxiangdong) [inline huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go:236]: pls do not set the password field before printing opts to the log file

Comment (hmbig2) [reply huaweicloud/services/dws/resource_huaweicloud_dws_cluster.go:236]: Ok, remove the log.printf

**Pull request merged**